### PR TITLE
Commandline flags

### PR DESCRIPTION
**Pull request by eddywine**
## Description

_Setting flags that will be called with the command Dotnet run for the executable

## Changes

* Changes to the APRS Connection library to take the server, callsign, filter, password 
* Console app to have default values whenever a flag is not set
* Options to set the different flags with their associated values for the application
* Adding different aliases for the filters

## Validation

* All the tests pass for the application
* The default flags will be used in the connection by running `dotnet run` 
* A given flag value will be used when set for example `dotnet run --c NOCALL --p -1`
* All the flags can be set for the connection to be made with the updated parameters
* Different aliases for the flags to be used for example (for password, the options are --p, --password, --pass)
* Aliases for server: --server, --s, --svr
* Aliases for the callsign: --callsign, --c, --cgn

### CODE REVIEW

File: src/APRSsharp/Program.cs [hunk 1]
```diff
@@ -24,21 +27,49 @@ public static void PrintTcpMessage(string tcpMessage)
         /// </summary>
         /// <param name="args"> The input arguments for the program i.e packets which will be strings.</param>
         public static void Main(string[] args)
+        {
+            // Create a root command with some options
+            var rootCommand = new RootCommand
+                {
+                new Option<string>(
+                    aliases: new string[] { "--password", "--p", "--pass" },
```
Comment (CBielstein) [inline src/APRSsharp/Program.cs:35]: I think we should try and follow the usual convention (at least in my experience, feel free to disagree) of two dashes for the full name (`--password`, like you have) and one dash for the short flag (`-p`).

This follows stuff like `git push --force` and `git push -f`

File: src/AprsISLibrary/AprsIsConnection.cs [hunk 2]
```diff
@@ -46,14 +46,12 @@ public AprsIsConnection(ITcpConnection tcpConnection)
         /// </summary>
         /// <param name="callsign">The users callsign string.</param>
         /// <param name="password">The users password string.</param>
+        /// <param name="server">The specified server to be connected.</param>
+        /// <param name="filter">The filter that is put for the connection.</param>
         /// <returns>An async task.</returns>
-        public async Task Receive(string? callsign, string? password)
+        public async Task Receive(string callsign, string password, string server, string filter)
```
Comment (CBielstein) [inline src/AprsISLibrary/AprsIsConnection.cs:52]: Can we leave these as optional? We want `AprsIsConnection` to support multiple use cases. The command line app is one, a future GUI app is another, or other users (open source, we don't know) as well. I appreciate your effort to consolidate the default values in to one spot in the RootCommand object.

I think a good way to reference the default values from two places is to create a static class in AprsIsConnection that holds these reused values. Then we can reference them from both spots. Something like

```csharp
public static class AprsIsConstants
{
    public const string DefaultCallsign = "N0CALL";
    public const string DefaultPassword = "-1";
    ...
}

**Review by CBielstein — APPROVED** [review]
Looks great! Thanks for making the updates!

File: README.md [hunk 3]
```diff
@@ -58,3 +58,9 @@ The resulting binary will be placed in
 `bin\Release\net6.0\win-x86\publish`.
 Alternatively, use `dotnet publish -c Release -o <outputfolder>` to specify the
 output directory.
+
+### Testing the project/application binary file
+
+To test the generated console application binary, go to the APRSsharp folder
+(`AprsSharp\src\APRSsharp`) and run the command `dotnet run`
```
[the remark this answers is not in the document]
Comment (eddywine) [reply README.md]: This is true, let me do that right away

**Pull request merged**